### PR TITLE
EP/CC: implement session binding for non-SPA's and improve default player control styling

### PR DESCRIPTION
**Pull request by OleksivO**
…reset after page's reload

### CODE REVIEW

File: src/embed-player.js [hunk 1]
```diff
@@ -388,21 +388,28 @@ export default class EmbedPlayer {
...
-                }
-            };
+        return new Promise(resolve => {
+            const castButtonContaner = document.querySelector(selector);
+            const castButton = document.createElement('google-cast-launcher');
+            castButtonContaner.appendChild(castButton);
+            if (chromecastReceiverAppId) {
+                window['__onGCastApiAvailable'] = (isAvailable) => {
+                    if (isAvailable && cast && cast.framework) {
+                        this.initializeCastApi(chromecastReceiverAppId);
```
Comment (jcarolus) [inline src/embed-player.js:398]: should the promise be resolved from here?

File: demo/video-cast.js [hunk 2]
```diff
@@ -21,9 +21,11 @@ import ChromecastControls from '../src/chromecast-controls.js';
 
     document.getElementById('video-button-start').addEventListener('click', playVideo);
     document.getElementById('video-button-stop').addEventListener('click', stopCastVideo);
-
-    player.setupChromecast('#cast-wrapper', chromecastReceiverAppId);
-    document.getElementById('cast-wrapper').style.display = 'unset';
+    player.setupChromecast('#cast-wrapper', chromecastReceiverAppId)
+        .then(() => {
+            const controls = new ChromecastControls(player.getCastPlayer(), player.getCastPlayerController());
+            document.getElementById('cast-wrapper').style.display = 'unset';
```
Comment (jcarolus) [inline demo/video-cast.js:27]: can the cast-wrapper style.display be unset in our library?

**Review by jcarolus — CHANGES_REQUESTED** [review]

File: demo/video-cast.js (hunk 2 above)
Comment (OleksivO) [reply demo/video-cast.js:27]: It's not part of the library but only our example. Other devs may not even hide it etc.

File: src/embed-player.js (hunk 1 above)
Comment (OleksivO) [reply src/embed-player.js:398]: done

**Review by jcarolus — APPROVED** [review]

**Pull request merged**